### PR TITLE
fix a lot of bugs uncovered by go 1.11

### PR DESCRIPTION
**Pull request by BenTheElder**
/cc @cjwagner

### CODE REVIEW

**Comment by BenTheElder** [comment]
weird mkpj integration failure:
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //prow/test:integration-mkpj
-----------------------------------------------------------------------------
[ERROR] Generated incorrect ProwJob YAML for config:
54,55c54
<   prev_report_states: {}
<   startTime: 2018-07-06T22:59:02Z
---
>   startTime: 2018-10-09T19:05:14Z
```
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/9747/pull-test-infra-bazel/20635/

File: prow/apis/prowjobs/v1/types.go [hunk 1]
```diff
@@ -249,7 +249,7 @@ type ProwJobStatus struct {
 
 	// PrevReportStates stores the previous reported prowjob state per reporter
 	// So crier won't make duplicated report attempt
-	PrevReportStates map[string]ProwJobState `json:"prev_report_states, omitempty"`
+	PrevReportStates map[string]ProwJobState `json:"prev_report_states,omitempty"`
```
Comment (krzyzacy) [inline prow/apis/prowjobs/v1/types.go]: hummmm because of this?
/shrug

**Comment by cjwagner** [comment]
One more: `prow/client/clientset/versioned/fake/clientset_generated.go:56: literal copies lock value from fakePtr: k8s.io/test-infra/vendor/k8s.io/client-go/testing.Fake`

**Comment by BenTheElder** [comment]
yeah that one is presumably generated code though... not sure how to best handle that yet

**Comment by BenTheElder** [comment]
I think this has been patched, we'll probably have to upgrade https://github.com/kubernetes/code-generator/commits/master/cmd/client-gen

**Comment by BenTheElder** [comment]
I updated the k8s deps from 1.10 to 1.11 which looks to have some relevant govet fixes in the generators tentatively.

I had to also update the json-iterator library xref: https://github.com/kubernetes/apimachinery/issues/46

**Comment by BenTheElder** [comment]
OK govet and gofmt are happy now.

File: prow/apis/prowjobs/v1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // +build !ignore_autogenerated
 
 /*
-Copyright 2018 The Kubernetes Authors.
+Copyright The Kubernetes Authors.
```
Comment (krzyzacy) [inline prow/apis/prowjobs/v1/zz_generated.deepcopy.go]: hummm so looks like we have updated the boilerplate :-)

Comment (BenTheElder) [reply prow/apis/prowjobs/v1/zz_generated.deepcopy.go]: ugh :man_facepalming:

File: hack/verify-govet.sh [hunk 3]
```diff
@@ -18,4 +18,4 @@ set -o nounset
 set -o pipefail
 set -o xtrace
 
-go vet $(go list ./... | grep -v vendor)
+go vet $(go list ./... | grep -v 'k8s.io/test-infra/testgrid')
```
Comment (spiffxp) [inline hack/verify-govet.sh]: can we drop a TODO or followup issue for testgrid?

Comment (BenTheElder) [reply hack/verify-govet.sh]: I guess. this one isn't really doable in any trivial fashion because the code doesn't even build with go @fejta

Comment (BenTheElder) [reply hack/verify-govet.sh]: govet breaks on importing a package that doesn't exist (because it's bazel generated)

Comment (BenTheElder) [reply hack/verify-govet.sh]: filed https://github.com/kubernetes/test-infra/issues/9749

File: prow/apis/prowjobs/v1/types.go (hunk 1 above)
Comment (BenTheElder) [reply prow/apis/prowjobs/v1/types.go]: yeah, fixed the test data

File: prow/apis/prowjobs/v1/zz_generated.deepcopy.go (hunk 2 above)
Comment (BenTheElder) [reply prow/apis/prowjobs/v1/zz_generated.deepcopy.go]: fixed by skipping generated files

**Comment by krzyzacy** [comment]
thanks for fighting the :fire:  :-)

/lgtm

**Pull request merged**